provisioner/puppet-masterless/provisioner.go

```diff
@@ -62,45 +65,48 @@ type Config struct {
 
 	// If true, packer will ignore all exit-codes from a puppet run
 	IgnoreExitCodes bool `mapstructure:"ignore_exit_codes"`
-
-	// The Guest OS Type (unix or windows)
-	GuestOSType string `mapstructure:"guest_os_type"`
 }
 
 type guestOSTypeConfig struct {
+	tempDir          string
 	stagingDir       string
 	executeCommand   string
 	facterVarsFmt    string
 	facterVarsJoiner string
 	modulePathJoiner string
 }
 
+// FIXME assumes both Packer host and target are same OS
```

Comment:
Will this code break if this assumption is not met?

Reply:
You can find out the host system by using runtime.GOOS (you can search the code for examples)

Reply:
Yes it will break. Even Windows->Windows will break since SYSTEMRoot on local and remote can easily be different. runtime.GOOS only tells me the OS of the machine running Packer, not what's on the other end of the provisioner. Am I wrong?

Reply:
No, but GuestOSType will tell us what is on the other side of the provisioner. So using them together should give you the full picture, right?

Reply:
it helps but 2 windows instances can have different %SYSTEMROOT% values. 'stagingDir' is particularly problematic. I'm pretty new at this. Not sure how to 'query' system attributes of the remote end. Even in Unix-land (/tmp) is not necessarily assured and we don't actually test existence nor writability/execute permissions.

Reply:
Yeah, it's an interesting problem. With most of the provisioners we try not to know anything about the guest system, use reasonable defaults/env vars where we can, and make sure that potential problem areas are configurable.

It may be worth it in the future to think about allowing users to directly customize the entire guestOSTypeConfig, though that's outside scope for this PR.

It looks to me like this issue already existed before you PR, and you're just adding a comment explaining that it's a problem. Is that accurate?

Reply:
yes it was just adding a heads-up comment.